### PR TITLE
Add emscripten_sleep() wrapper to libc-emcc

### PR DESCRIPTION
**Pull request by bnason-nf**

### CODE REVIEW

**Comment by bnason-nf** [comment]
There was a failure in one of the checks that seems unrelated to the change.

File: core/iwasm/libraries/libc-emcc/libc_emcc_wrapper.c [hunk 1]
```diff
@@ -495,6 +495,12 @@ emscripten_notify_memory_growth_wrapper(wasm_exec_env_t exec_env, int i)
     (void)i;
 }
 
+static void
+emscripten_sleep_wrapper(wasm_exec_env_t exec_env, int timeout_ms)
```
Comment (wenyongh) [inline core/iwasm/libraries/libc-emcc/libc_emcc_wrapper.c:499]: Is the time unit ms? `man 3 sleep` in linux shows:
```C
NAME
       sleep - sleep for a specified number of seconds

SYNOPSIS
       #include <unistd.h>

       unsigned int sleep(unsigned int seconds);
```

File: core/iwasm/libraries/libc-emcc/libc_emcc_wrapper.c [hunk 2]
```diff
@@ -495,6 +495,12 @@ emscripten_notify_memory_growth_wrapper(wasm_exec_env_t exec_env, int i)
...
+{
+    usleep(timeout_ms * 1000);
```
Comment (wenyongh) [inline core/iwasm/libraries/libc-emcc/libc_emcc_wrapper.c:501]: integer overflow may occur in `timeout_ms * 1000`, had better check it?
```C
    if (timeout > UINT32_MAX /1000)
        usleep(UINT32_MAX);
    else
        usleep(timeout_ms * 1000);
```

File: core/iwasm/libraries/libc-emcc/libc_emcc_wrapper.c (hunk 1 above)
Comment (bnason-nf) [reply core/iwasm/libraries/libc-emcc/libc_emcc_wrapper.c:499]: Yes this is confusing, but I believe the implementation is correct. Please see the emscripten documention here:

https://emscripten.org/docs/api_reference/emscripten.h.html#sleeping

File: core/iwasm/libraries/libc-emcc/libc_emcc_wrapper.c (hunk 2 above)
Comment (bnason-nf) [reply core/iwasm/libraries/libc-emcc/libc_emcc_wrapper.c:501]: Thanks, your comments made me realize that the implementation needs to be more robust in several ways, so I've rewritten it.

Comment (bnason-nf) [reply core/iwasm/libraries/libc-emcc/libc_emcc_wrapper.c]: Sorry about that, it's pushed now.

**Review by wenyongh — APPROVED** [review]
LGTM

**Comment by TianlongLiang** [comment]
LGTM

**Pull request merged**